docs/guide/interactions/views.rst

```diff
@@ -0,0 +1,483 @@
+
+Additionally, since this view is persistent we need to specify a :attr:`~discord.ui.Select.custom_id` for our 
+:class:`~discord.ui.Select` component, which is used to identify the component when a user interacts with it.
+As we could have multiple role selectors, it seems fitting to use the ID of the message the view is attached
+to as part of the ``custom_id``.
+
+After adding these details, our code will look something like this:
+
+.. code-block:: python
+    :emphasize-lines: 2,5-9
+
+    class RoleSelector(discord.ui.View):
+        def __init__(self, message_id: int, roles: List[discord.Role]) -> None:
+            super().__init__(timeout=None)
+
+            self.selector.custom_id = f'role_selector:{message_id}'
+
+            self.roles: List[discord.Role] = roles
+            for role in roles:
+                self.selector.add_option(label=role.name, value=str(role.id))
+
+        @discord.ui.select(
+            placeholder='Select a role',
+            min_values=1,
+            max_values=1,
+        )
+        async def selector(
+            self,
+            interaction: discord.Interaction,
+            component: discord.ui.Select['RoleSelector'],
+        ) -> None:
+            raise NotImplementedError
+
+We also need to add a body to our ``selector`` callback function, which will assign the selected role to the user.
+This is fairly simple to do, when the callback is invoked we can access the options the user selected via the
+:attr:`~discord.ui.Select.options` attribute. This holds a list of values, but since we limited the number of values to `1`,
+we can just access the first element directly. We can then use sets to determine what roles the user already has, 
+and then override the member's roles with :meth:`Member.edit <discord.Member.edit>`.
+
+Our function body might look like this:
+
+.. code-block:: python
+    :emphasize-lines: 21-25
+
+    class RoleSelector(discord.ui.View):
+        def __init__(self, message_id: int, roles: List[discord.Role]) -> None:
+            super().__init__(timeout=None)
+
+            self.selector.custom_id = f'role_selector:{message_id}'
+
+            self.roles: List[discord.Role] = roles
+            for role in roles:
+                self.selector.add_option(label=role.name, value=str(role.id))
+
+        @discord.ui.select(
+            placeholder='Select a role',
+            min_values=1,
+            max_values=1,
+        )
+        async def selector(
+            self,
+            interaction: discord.Interaction,
+            component: discord.ui.Select['RoleSelector'],
+        ) -> None:
+            role_id = int(component.values[0])
+            role = discord.Object(id=role_id, type=discord.Role)
+            roles = set(interaction.user.roles) - set(self.roles) | {role}
+            await interaction.user.edit(roles=roles)
+            await interaction.response.defer()
```

Comment:
Instead of deferring the interaction and not acknowledging the response, you should send an message response instead. The contents of the message should tell the user that their role has been changed to the role that they have selected, and this should be ephemeral.